src/kafka/base.rs

```diff
@@ -11,6 +14,60 @@ pub trait AlertProducer {
     async fn produce(&self, topic: Option<String>) -> Result<i64, Box<dyn std::error::Error>>;
 }
 
+pub async fn ensure_kafka_topic_partitions(
+    bootstrap_servers: &str,
+    topic_name: &str,
+    expected_nb_partitions: usize,
+) -> Result<usize, Box<dyn std::error::Error>> {
+    let producer: FutureProducer = ClientConfig::new()
+        .set("bootstrap.servers", bootstrap_servers)
+        .create()?;
+
+    let admin_client: AdminClient<DefaultClientContext> = ClientConfig::new()
+        .set("bootstrap.servers", bootstrap_servers)
+        .create()?;
+
+    // Fetch all topics
+    let metadata = producer
+        .client()
+        .fetch_metadata(None, std::time::Duration::from_secs(5))?;
+    let existing_topics = metadata
+        .topics()
+        .iter()
+        .map(|t| t.name().to_string())
+        .collect::<Vec<String>>();
+
+    let topic_exists = existing_topics.contains(&topic_name.to_string());
+
+    let nb_partitions = if topic_exists {
+        info!("Topic {} exists, using existing partitions", topic_name);
+        metadata
+            .topics()
+            .iter()
+            .find(|t| t.name() == topic_name)
```

Comment:
There may be an opportunity to streamline this function. `find` returns `Some` if the topic is found and `None` if it doesn't exist. If I'm thinking about this right, this means we can get rid of the `existing_topics` part above.